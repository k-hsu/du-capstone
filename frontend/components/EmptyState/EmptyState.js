import React from "react";
import Flex from "../Flex/Flex";
import Text from "../Text/Text";

const EmptyStateElement = ({ type = "loading" }) => {
  return (
    <Flex width="100%" justifyContent="center">
      <Text>{type === "loading" ? "loading..." : "An error has occured"}</Text>
    </Flex>
  );
};

const EmptyState = ({ loading }) => {
  return loading ? (
    <EmptyStateElement type="loading" />
  ) : (
    <EmptyStateElement type="error" />
  );
};

export default EmptyState;
