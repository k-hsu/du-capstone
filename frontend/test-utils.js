import { render } from "@testing-library/react";
import userEvent from "@testing-library/user-event";
import { renderHook } from "@testing-library/react-hooks";

function setup(jsx) {
  return {
    user: userEvent.setup(),
    ...render(jsx),
  };
}

export * from "@testing-library/react";
export { renderHook, setup };
