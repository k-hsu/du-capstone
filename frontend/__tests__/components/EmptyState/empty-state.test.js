import React from "react";
import { render, screen, renderer } from "../../../test-utils";
import EmptyState from "../../../components/EmptyState/EmptyState";

describe("EmptyState", () => {
  it("should render empty state component of type loading", () => {
    render(<EmptyState loading />);
    expect(screen.getByText("loading...")).toBeInTheDocument();
  });
  it("should render empty state component of type error", () => {
    render(<EmptyState />);
    expect(screen.getByText("An error has occured")).toBeInTheDocument();
  });
  it("should render empty state component styles of type loading", () => {
    const component = renderer.create(<EmptyState loading />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
  it("should render empty state component styles of type error", () => {
    const component = renderer.create(<EmptyState />);
    const tree = component.toJSON();
    expect(tree).toMatchSnapshot();
  });
});
