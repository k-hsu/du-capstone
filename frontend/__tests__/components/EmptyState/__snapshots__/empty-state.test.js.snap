// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`EmptyState should render empty state component styles of type error 1`] = `
<div
  className="css-q1uf87"
  direction="row"
  width="100%"
>
  <text
    className="css-rsymv"
    fontFamily="IBM Plex Sans"
  >
    An error has occured
  </text>
</div>
`;

exports[`EmptyState should render empty state component styles of type loading 1`] = `
<div
  className="css-q1uf87"
  direction="row"
  width="100%"
>
  <text
    className="css-rsymv"
    fontFamily="IBM Plex Sans"
  >
    loading...
  </text>
</div>
`;
